lib/models/purchasable.dart

```diff
@@ -0,0 +1,7 @@
+
+import 'presented_offering_context_wrapper.dart';
+
+/// Interface for items that can be purchased
+abstract class Purchasable {
```

Comment:
I thought about the best way of doing this in dart... The other option I thought was to use sealed classes, but we're trying to avoid sealed classes to make it extensible in the future without causing breaking changes.

Also thought about creating multiple constructors of the `PurchaseParams` object, each receiving a Package/StoreProduct/SubscriptionOption, but that caused more duplication IMO...

Reply:
That's correct, but as you said, it would fail later when trying to purchase it

Reply:
Great! I think it's fine 👍

Reply:
One other thing, I also thought about following the same pattern we do with `PurchasesConfiguration`... but felt to me this is better because it ends up with an immutable object. In `PurchasesConfiguration`, all parameters are mutable, which I didn't fully like... on the other hand, it's not as consistent to do something different. Lmk if you have any concerns over this @RevenueCat/coresdk